pkg/parse/run.go

```diff
@@ -238,6 +251,14 @@ func run(ctx context.Context, p Parser, trigger string, state *reconcilerState)
 // It also updates the .status.rendering and .status.source fields.
 func read(ctx context.Context, p Parser, trigger string, state *reconcilerState, sourceState sourceState) status.MultiError {
 	hydrationStatus, sourceStatus := readFromSource(ctx, p, trigger, state, sourceState)
+	if p.options().renderingEnabled != hydrationStatus.requiresRendering {
+		// the reconciler is misconfigured. set the annotation so that the reconciler-manager
+		// will recreate this reconciler with the correct configuration.
+		if err := p.setRenderingRequired(ctx, hydrationStatus.requiresRendering); err != nil {
```

Comment:
Can we unify and call this setRequiresRendering instead of setRenderingRequired, given the name of the annotation and the status field? Especially when there's `RenderingRequired` const that has slightly different meaning.

Reply:
Yeah seems reasonable

Reply:
done